include/morphio/section_iterators.hpp

```diff
@@ -56,8 +56,9 @@ std::shared_ptr<SectionT> getParent(const std::shared_ptr<SectionT>& current)
 
 namespace morphio {
 
-template<typename SectionT, typename MorphologyT>
-class breadth_iterator_t {
+template <typename SectionT, typename MorphologyT>
+class breadth_iterator_t : public std::iterator<std::input_iterator_tag, const SectionT>
```

Comment:
I thought `std::iterator`  was on the way to deprecation.  I prefer to set the iterator traits manually, as it's more explicit, imo - don't have to remember the order of `std::iterator`.

Reply:
Thank you for your remark. Indeed one should not use `std::iterator`. Change 45aba85 fixes that.

Reply:
Cool.
@wizmer is back, so I will leave the rest of the review in his hands.  I read a bunch of it last week, and it looked good.  Thanks for the effort - I know how tedious this can be.